### PR TITLE
Action Codes and Goal Units

### PR DESCRIPTION
**Pull request by cKellyDesign**
Closes: #498 
Closes: #563

### CODE REVIEW

File: src/store/ducks/goals.ts [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
+import { PlanActionCodes } from '../../configs/settings';
 import { FlexObject } from '../../helpers/utils';
 
 /** the reducer name */
 export const reducerName = 'goals';
 
 /** interface for goal Object */
 export interface Goal {
-  action_code: string;
+  action_code: typeof PlanActionCodes[number];
```
Comment (moshthepitt) [inline src/store/ducks/goals.ts]: I think this one is already defined here https://github.com/onaio/reveal-frontend/blob/b0ea6c03e5479537a782aa5340b757087d3b5214/src/containers/forms/PlanForm/types.ts#L29

File: src/constants.tsx [hunk 2]
```diff
@@ -96,7 +97,7 @@ export const STRUCTURE_LAYER = 'structure-layer';
 export type STRUCTURE_LAYER = typeof STRUCTURE_LAYER;
 export const PROGRESS = 'Progress';
 export type PROGRESS = typeof PROGRESS;
-export const STRUCTURES = 'structure(s)';
+export const STRUCTURES = `${STRUCTURES_UNIT_TYPE}(s)`;
```
Comment (moshthepitt) [inline src/constants.tsx]: Will this switch everything to be the value of the env defined?  It seems for #498 we want just the last indicator to say rooms while the others say structures.

Comment (cKellyDesign) [reply src/constants.tsx]: oh hmm, i thought one of the countries measured sprayable units as rooms, let me get clarification in the ticket

File: src/store/ducks/goals.ts (hunk 1 above)
Comment (cKellyDesign) [reply src/store/ducks/goals.ts]: oh cool, will switch

File: src/constants.tsx (hunk 2 above)
Comment (cKellyDesign) [reply src/constants.tsx]: thanks for the explanation @moshthepitt changes made

**Comment by moshthepitt** [comment]
Nice!  What kind of test can we add for this, @cKellyDesign ?

**Comment by cKellyDesign** [comment]
Haaaaa, of the Unit variety! Adding now :)

**Comment by moshthepitt** [comment]
Nice :)

**Pull request merged**